### ISSUE TITLE
Proxy exec problem

### ISSUE THREAD

**Issue by sanjesss** [lensapp/lens#8119]
Hi, I have a problem with exec inside the feed.
I have a kubrnetes cluster on bare metal. When connecting directly to it on host master node via Lens, exec inside pod works fine.
I can also connect to this cluster via HAproxy, in this case exec inside the pod via Lens does not work.
HAproxy works in TCP mode proxying all requests from *:6443 to kube-master:6443 without any processing.
At the same time through k9s and kubectl exec inside the feed works both directly and through HAproxy. 
The same situation happens with exec inside a node.
I suspect that Lens uses some additional port for exec, which I also need to proxy.

**Environment (please complete the following information):**
- Lens Version:
Lens: 2024.10.171859-latest
Electron: 31.6.0
Chrome: 126.0.6478.234
Node: 20.17.0
- OS: Windows 11, MacOS Sequoia
- Installation method (e.g. snap or AppImage in Linux): *.dmg and *.exe form lens website

**Logs exec into pod:**
```
PS C:\Users\sanje> kubectl exec -i -t -n namespace pod-name-797fcdf95b-gpfl1 -c  pod-name -- sh -c "clear; (bash || ash || sh)"
Error from server:
```
**Logs exec into node:**
```
Error from server:
Terminal will auto-close in 15 seconds ...
```

**HAproxy config:**
```
  frontend kubernetes_api
    mode tcp
    bind *:6443
    tcp-request inspect-delay 5s
    tcp-request content accept if { req_ssl_hello_type 1 }
    acl host_cluster0 req_ssl_sni -i cluster0.eu
    acl host_cluster1 req_ssl_sni -i cluster1.eu
    acl host_cluster2 req_ssl_sni -i cluster2.eu
    use_backend kubeapi_cluster0 if host_cluster0
    use_backend kubeapi_cluster1 if host_cluster1
    use_backend kubeapi_cluster2 if host_cluster2

backend kubeapi_cluster0
    mode tcp
    balance roundrobin
    server cluster0_master master.cluster0:6443 check

backend kubeapi_cluster1
    mode tcp
    balance roundrobin
    server cluster1_master master.cluster1:6443 check

backend kubeapi_cluster2
    mode tcp
    balance roundrobin
    server cluster2_master master.cluster2:6443 check
```

**Comment by Christiaanvdl** [comment]
Hello @sanjesss , Thank you for reaching out i have contacted our development team regarding your issue .

**Comment by dechegaray** [comment]
> Hi, I have a problem with exec inside the feed. I have a kubrnetes cluster on bare metal. When connecting directly to it on host master node via Lens, exec inside pod works fine. I can also connect to this cluster via HAproxy, in this case exec inside the pod via Lens does not work. HAproxy works in TCP mode proxying all requests from *:6443 to kube-master:6443 without any processing. At the same time through k9s and kubectl exec inside the feed works both directly and through HAproxy. The same situation happens with exec inside a node. I suspect that Lens uses some additional port for exec, which I also need to proxy.
> 
> **Environment (please complete the following information):**
> 
> * Lens Version:
>   Lens: 2024.10.171859-latest
>   Electron: 31.6.0
>   Chrome: 126.0.6478.234
>   Node: 20.17.0
> * OS: Windows 11, MacOS Sequoia
> * Installation method (e.g. snap or AppImage in Linux): *.dmg and *.exe form lens website
> 
> **Logs exec into pod:**
> 
> ```
> PS C:\Users\sanje> kubectl exec -i -t -n namespace pod-name-797fcdf95b-gpfl1 -c  pod-name -- sh -c "clear; (bash || ash || sh)"
> Error from server:
> ```
> 
> **Logs exec into node:**
> 
> ```
> Error from server:
> Terminal will auto-close in 15 seconds ...
> ```
> 
> **HAproxy config:**
> 
> ```
>   frontend kubernetes_api
>     mode tcp
>     bind *:6443
>     tcp-request inspect-delay 5s
>     tcp-request content accept if { req_ssl_hello_type 1 }
>     acl host_cluster0 req_ssl_sni -i cluster0.eu
>     acl host_cluster1 req_ssl_sni -i cluster1.eu
>     acl host_cluster2 req_ssl_sni -i cluster2.eu
>     use_backend kubeapi_cluster0 if host_cluster0
>     use_backend kubeapi_cluster1 if host_cluster1
>     use_backend kubeapi_cluster2 if host_cluster2
> 
> backend kubeapi_cluster0
>     mode tcp
>     balance roundrobin
>     server cluster0_master master.cluster0:6443 check
> 
> backend kubeapi_cluster1
>     mode tcp
>     balance roundrobin
>     server cluster1_master master.cluster1:6443 check
> 
> backend kubeapi_cluster2
>     mode tcp
>     balance roundrobin
>     server cluster2_master master.cluster2:6443 check
> ```

There is an open issue related to this topic. The problem is that Lens uses an internal proxy to route requests to the Kube API of each cluster, and when other proxies are put in front of it (i.e. HAProxy, Tailscale, etc), ours is disrupted and cannot execute appropriately node shells, pod shells or port-forwards. 

We are currently working on a solution for this. Please stay tune for an update that will patch this issue in the next couple of weeks.

**Comment by sanjesss** [comment]
@dechegaray 
Hi, I checked version 2024.11.261604 and the bug is not fixed yet. When is a fix planned?

**Comment by dechegaray** [comment]
> @dechegaray Hi, I checked version 2024.11.261604 and the bug is not fixed yet. When is a fix planned?

@sanjesss That's correct, the team is still working on this (it wasn't scoped for `2024.11.261604`). Most likely it will be available in the next release we do (maybe before the end of the year), I will notify you as soon as we have a concrete date

**Comment by sm1lexops** [comment]
this error/bug is most often repeated [can't pod exec issues/8113](https://github.com/lensapp/lens/issues/8113) 
I counted five issues of the same kind
proxy it isn't a problem

**Comment by sanjesss** [comment]
@sm1lexops I have this problem only if the haproxy is in front of KubeAPI. In all other cases everything works normally. So it is not the same thing.

**Comment by dechegaray** [comment]
> @sm1lexops I have this problem only if the haproxy is in front of KubeAPI. In all other cases everything works normally. So it is not the same thing.

@sanjesss we just released a new Beta version that bypasses **by default** the internal proxy we have when executing commands (node/pod shell, port forwards, helm, etc). Please give this version a try when possible to confirm if this definitely fixes the problem with your haproxy setup, etc. Announcement -> https://forums.k8slens.dev/t/5306

![Screenshot 2024-12-20 at 11 01 41 AM](https://github.com/user-attachments/assets/0c96aa0d-49b4-4649-ac8a-9b2f88e6e3a0)

**Comment by savinov** [comment]
Same issue, but using VPN connection to access k8s cluster: can't open k8s container terminal from Lens, but same `exec kubectl exec -i -t -n ... -- sh -c "clear; (bash || ash || sh)"` command works fine from OS terminal.
Beta version fixed the problem, thanks!
https://forums.k8slens.dev/t/5306

**Comment by dechegaray** [comment]
@savinov just in case, we just released a new **stable** version this week that contains these changes (https://forums.k8slens.dev/t/5377). Feel free to upgrade if you don't want to keep using the `beta` anymore. Just make sure to turn **ON** the bypass in `Preferences > Poxy`

**Comment by sm1lexops** [comment]
after updating to latest release 2025.1.161916-latest
all is working as expected

**Comment by aaronwng** [comment]
This is still an issue with wireguard vpn
